apps/ledger-live-mobile/src/screens/Manager/Device/DeviceLanguage.tsx

```diff
@@ -69,29 +70,23 @@ const DeviceLanguage: React.FC<Props> = ({
   const refreshDeviceLanguage = useCallback(() => {
     track("Page Manager LanguageInstalled", { selectedLanguage });
     onLanguageChange();
-  }, [selectedLanguage]);
+  }, [selectedLanguage, onLanguageChange]);
 
   return (
     <>
-      <Flex flex={1} flexDirection="row" alignItems="center">
-        <Icons.LanguageMedium size={24} color="neutral.c80" />
-        <Text ml={3} flex={1} variant="bodyLineHeight" color="neutral.c80">
-          {t("deviceLocalization.language")}
-        </Text>
-        {availableLanguages.length ? (
-          <Link
-            onPress={pendingInstalls ? undefined : openChangeLanguageModal}
-            type="color"
-            Icon={Icons.ChevronRightMedium}
-          >
-            {t(`deviceLocalization.languages.${currentDeviceLanguage}`)}
-          </Link>
-        ) : (
-          <Text>
-            {t(`deviceLocalization.languages.${currentDeviceLanguage}`)}
-          </Text>
-        )}
-      </Flex>
+      <DeviceOptionRow
+        Icon={Icons.LanguageMedium}
+        label={t("deviceLocalization.language")}
+        onPress={pendingInstalls ? undefined : openChangeLanguageModal}
+        linkLabel={t(`deviceLocalization.languages.${currentDeviceLanguage}`)}
+        right={
+          availableLanguages.length ? undefined : (
```

Comment:
wait what happens if `availableLanguages.length === 0` ? the default `<Link>` component from `DeviceOptionRow` is displayed with the value of `linkLabel`? That was not the behavior before no ? We want a `<Text>` in the case there is no available language ?